src/lib/memoise.ts

```diff
@@ -0,0 +1,16 @@
+export const memoise = <T>(fn: () => Promise<T>): [() => void, () => Promise<T>] => {
+    let resp: T | undefined;
+
+    const retFn = async () => {
+        if (resp) {
```

Comment:
Should we be more specific here and check for not `undefined`? I think `null`/`false` could both be valid memoizable return values.

Reply:
I've falled victim to Javascript again 🤦‍♂